commit 80e179128cfd78d95cdebf7195fd21299e7931b6
Author: Willy Tarreau <w@1wt.eu>
Date:   Sat Apr 28 07:18:15 2018 +0200

    BUG/MINOR: config: disable http-reuse on TCP proxies
    
    Louis Chanouha reported an inappropriate warning when http-reuse is
    present in a defaults section while a TCP proxy accidently inherits
    it and finds a conflict with other options like the use of the PROXY
    protocol. To fix this patch removes the http-reuse option for TCP
    proxies.
    
    This fix needs to be backported to 1.8, 1.7 and possibly 1.6.
    
    (cherry picked from commit 46deab6e64bfda7211b7c3199ad01f136141c86f)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 5a460381..63d2de58 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -8702,6 +8702,9 @@ out_uri_auth_compat:
 			}
 #endif
 
+			if ((curproxy->mode != PR_MODE_HTTP) && (curproxy->options & PR_O_REUSE_MASK) != PR_O_REUSE_NEVR)
+				curproxy->options &= ~PR_O_REUSE_MASK;
+
 			if ((curproxy->options & PR_O_REUSE_MASK) != PR_O_REUSE_NEVR) {
 				if ((curproxy->conn_src.opts & CO_SRC_TPROXY_MASK) == CO_SRC_TPROXY_CLI ||
 				    (curproxy->conn_src.opts & CO_SRC_TPROXY_MASK) == CO_SRC_TPROXY_CIP ||
